### ISSUE TITLE
Add new props to the Hr element

### ISSUE THREAD

**Issue by khushal87** [mantinedev/mantine#28]
Currently, the `Hr` component has only three props as `variant, size, and color`. We can have some more useful props such as inset(left, right, middle), orientation(vertical, horizontal), subHeader, and width. 

Note: I am considering Hr in this package to be a standard `Divider`

Progress:

- [x] Inset
- [x] Orientation
- [x] subHeader
- [x] subHeaderStyle
- [x] Docs
- [x] Tests
- [x] Stories

**Comment by khushal87** [comment]
Hello admins! I want to work on this issue if it seems relevant to you all. Thank you. @cstrat @rtivital @srezanova

**Comment by rtivital** [comment]
Hi @khushal87 sounds good!
Can you describe what subHeader prop will do?
Also please make sure that component works fine with [Menu](https://mantine.dev/core/menu/) and [Card](https://mantine.dev/core/card/) after the changes (in horizontal orientation only), since these components have some context styles for Hr.

**Comment by khushal87** [comment]
@rtivital Thanks, sure I will check the following. For subheaders, you can refer https://material.io/components/dividers#usage.
Also, can you explain what is the realtime testing method you are using to test the changes in the package components visually, this will improve my workflow.

**Comment by rtivital** [comment]
Yep, subHeader looks great, let's add it.
To get started locally:
- fork this repo
- clone your fork
- install dependencies with `yarn`
- run storybook – `npm run storybook` – we use storybook for local development, add stories that you need to test new features, don't worry on how it looks in storybook – it's not published anywhere
- don't forget to add props descriptions and tests
- after you finish adding new features run `npm run docs` – it will start local documentation server
- Add docs and demos to Hr component
- Done! Run npm test to check if everything is fine and send a PR

Here is a link to contributing guidelines – https://mantine.dev/pages/contribute/ with more in depth review

**Comment by khushal87** [comment]
Thanks a lot, @rtivital. I will soon raise a PR for the same.

**Comment by rtivital** [comment]
@khushal87 sorry, I forgot one thing, when you will create a PR, please request a merge to next-1.1.0 as this feature is planned to release in minor release

**Comment by cstrat** [comment]
> Thank you. @cstrat @rtivital @srezanova

Don't include me in that list, I've just helped fix some typos. Hoping I can contribute more though!

**Comment by khushal87** [comment]
@rtivital sure! 😄

**Comment by khushal87** [comment]
Hey, @rtivital I was just running the storybook locally and facing this problem here. Can you help?

```
ERROR in ./configuration/storybook/generated-stories-entry.js
Module not found: Error: Can't resolve 'C:Usershrithmantinesrc' in 'C:\Users\hrith\mantine\configuration\storybook'        
 @ ./configuration/storybook/generated-stories-entry.js 7:37-127
 @ multi ./node_modules/@storybook/core-server/dist/cjs/globals/polyfills.js ./node_modules/@storybook/core-server/dist/cjs/globals/globals.js ./configuration/storybook/storybook-init-framework-entry.js ./node_modules/@storybook/addon-docs/dist/esm/frameworks/common/config.js-generated-other-entry.js ./node_modules/@storybook/addon-docs/dist/esm/frameworks/react/config.js-generated-other-entry.js ./node_modules/@storybook/addon-actions/dist/esm/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-actions/dist/esm/preset/addArgs.js-generated-other-entry.js ./node_modules/@storybook/addon-backgrounds/dist/esm/preset/addDecorator.js-generated-other-entry.js ./node_modules/@storybook/addon-backgrounds/dist/esm/preset/addParameter.js-generated-other-entry.js ./configuration/storybook/preview.js-generated-config-entry.js ./configuration/storybook/generated-stories-entry.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined 
```

**Comment by rtivital** [comment]
Let me check, I haven't tried on windows yet

**Comment by khushal87** [comment]
There's some path issue depending on the operating system according to me.

**Comment by rtivital** [comment]
Sorry, I've spent half an hour debugging this on windows 10, and nothing seems to work, I can propose only to skip the storybook part and develop straight in docs, it also has HMR and should work on windows

**Comment by khushal87** [comment]
Okay, i will go through this issue, if I find the solution I will propose a fix in the meantime @rtivital

**Comment by rtivital** [comment]
Okay, sounds good

**Comment by khushal87** [comment]
Hey @rtivital I made this working by this change in the ./configuration/storybook/main.js

```
module.exports = {
  stories: [],
  addons: ['@storybook/addon-essentials'],
  webpackFinal: async (config) => {
    config.resolve = {
      ...config.resolve,
      plugins: [
        ...(config.resolve.plugins || []),
        new TsconfigPathsPlugin({
          extensions: ['.ts', '.tsx', '.js'],
          configFile: path.join(__dirname, '../../tsconfig.json'),
        }),
      ],
    };

    // Turn off docgen plugin as it breaks bundle with displayName
    config.plugins.pop();

    return config;
  },
};
```

I changes stories to []. Can you check if this works on other OS too, if this will work I will raise a PR and this would probably be a fix.

**Comment by rtivital** [comment]
Well, that fixes the problem but also removes all stories 😄 , the value for stories is required for storybook to work

**Comment by khushal87** [comment]
Exactly, so the next steps are adding a suitable path here to fetch all the stories, let me see if I can fix this too.

**Comment by khushal87** [comment]
@rtivital [path.resolve(__dirname, '../../src/**/*.story.@(ts|tsx)').replace(/\\/g, '/')] this will work fine in every OS in my opinion can I raise a PR for it?

**Comment by rtivital** [comment]
Yep, just checked, it works on MacOS:
```
stories: [path.resolve(__dirname, '../../src/**/*.story.@(ts|tsx)').replace(/\//g, '/')],
```

Please create the PR

**Comment by khushal87** [comment]
Thanks @rtivital

**Comment by khushal87** [comment]
@rtivital A quick doubt. Why is the default Hr variant 'dashed'? I believe the most suitable would be 'solid'

**Comment by rtivital** [comment]
I just always use dashed variant in my projects and never solid,but now that you add more features, lets change it to solid

**Comment by khushal87** [comment]
Okay. Thank you.

**Comment by khushal87** [comment]
@rtivital I believe that `width` is a better name instead of `size` for Hr and Dividers. What is your opinion? Should I change it?

**Comment by rtivital** [comment]
No let's leave size – it does not depend on the context. In vertical orientation size means width in horizontal – height. It's consistent with the rest of API

**Comment by khushal87** [comment]
Okay, got it. I am facing some issues with the vertical orientation styles, it is not coming up well. Let's see if different approaches work. Apart from that, everything is good to go. 💯

**Comment by rtivital** [comment]
Resolved in [1.1.0](https://github.com/mantinedev/mantine/releases/tag/1.1.0)
